source/opt/function.cpp

```diff
@@ -78,12 +78,19 @@ void Function::ForEachParam(const std::function<void(const Instruction*)>& f,
 }
 
 std::ostream& operator<<(std::ostream& str, const Function& func) {
-  func.ForEachInst([&str](const ir::Instruction* inst) {
-    str << *inst;
+  str << func.PrettyPrint();
+  return str;
+}
+
+std::string Function::PrettyPrint(uint32_t options) const {
+  std::string str;
+  ForEachInst([&str, options](const ir::Instruction* inst) {
+    str += inst->PrettyPrint(options);
```

Comment:
This will result in a lot of memory reallocations.
Streaming is much more preferable.